File: егэ/16 /якласс /02.cpp
#include <iostream>

int f(int n) {
	if (n == 1) return 2;
	if (n > 1 && n % 2 == 0) return 2 * n + f(n - 3);
	if (n > 1 && !(n % 2 == 0)) return 2 * (f(n + 1));
}

int main() {
	std::cout << f(13); // ans - 1592
	return 0;
}

File: егэ/8 /якласс /02.py
from itertools import permutations
k = 0

for x in permutations("ФУНКЦИЯ"):
    s = "".join(x)
    if s[0] != "Я" and "ЯК" not in s and "ЯН" not in s:
        k += 1
print(k)
# ans - 3120
